km/km_management.c

```diff
@@ -85,54 +94,86 @@ static void* mgt_main(void* arg)
       }
 
       // let them know what happened.
-      ssize_t bw = send(nfd, &mgmtreply, sizeof(mgmtreply), MSG_NOSIGNAL);
+      bw = send(nfd, &mgmtreply, sizeof(mgmtreply), MSG_NOSIGNAL);
       if (bw != sizeof(mgmtreply)) {
-         km_warn("send mgmt reply failed, byteswritten %ld", bw);
+         km_warn("send mgmt reply failed, byteswritten %ld, expected %d", bw, sizeof(mgmtreply));
       }
       close(nfd);
+      if (needunblock != 0) {
+         // We need to send the reply before potentially shutting down the payload threads.
+         km_snapshot_unblock();
+      }
+      if (mgmtreply.request_status == 0 && mgmtrequest.requests.snapshot_req.live == 0) {
+         // Payload threads are terminating, this thread doesn't need to receive any more mgmt requests.
+         break;
+      }
    }
    return NULL;
 }
 
-void km_mgt_init(char* path)
+void km_mgt_fini(void)
 {
-   if (path == NULL) {
+   unlink(addr.sun_path);
+   if (sock == -1) {
       return;
    }
+   kill_thread = 1;
```

Comment:
not needed is `sock == -1`?

Reply:
Im not sure what you are asking.
Do you think sock could be some other negative value than -1?
Do you think we are not setting sock to -1 when we should be?